scripts/parse_nagim.py

```diff
@@ -793,21 +1233,103 @@ def get_sequence_meta(self, sample_id: str, row: GroupedRow) -> Dict[str, Any]:
-        _call(f'gsutil ls "{source_bucket}/*.{ending_to_search}" >> {output_path}')
+        try:
+            _call(f'gsutil ls "{source_bucket}/*.{ending_to_search}" >> {output_path}')
+        except subprocess.CalledProcessError:
+            pass
     with open(output_path) as f:
         return [line.strip() for line in f.readlines() if line.strip()]
 
 
+@cli.command()
+@click.option('--tmp-dir', 'tmp_dir')
+@click.option('--dry-run', 'dry_run', is_flag=True)
+def move_uploads(
+    tmp_dir: str,
+    dry_run: bool,
+):
+    """
+    Assuming data is transferred to the upload buckets with transfer(),
+    and samples are populated with parse(), move data from upload buckets
+    to main and archive buckets, and update the SM analysis records.
+    """
+    if not tmp_dir:
+        tmp_dir = tempfile.gettempdir()
+
+    namespace = 'main'
+
+    # Listing files that exist in the upload bucket.
+    samples = _build_sample_list(tmp_dir, namespace, overwrite=False)
+
+    hbatch = setup_batch(
+        title='Process NAGIM uploads',
+        keep_scratch=False,
+        tmp_bucket=f'cpg-{NAGIM_PROJ_ID}-{namespace}-tmp',
+        analysis_project_name=NAGIM_PROJ_ID,
+    )
+
+    for proj in PROJECT_ID_MAP.values():
+        stack = _get_stack_id(proj)
+        smdb_proj = _get_sm_proj_id(proj, namespace)
+
+        def _move(j, file):
+            j.command(
+                f"""\
+if gsutil -q stat "{file.moved_path}"
+then
+    echo "target file already exists"
+else
+    if gsutil -q stat "{file.upload_path}"
+    then
+        echo "source file exists, target does not, safely moving"
+        gsutil mv "{file.upload_path}" "{file.moved_path}"
+    else
+        echo "source file does not exist"
+        exit 1
+    fi
+fi"""
+            )
+
+        logger.info(f'{proj}: adding jobs to move CRAM uploads')
+        new_bucket = SOURCES['CRAM'].get_final_bucket(stack, namespace)
+        for s in [s for s in samples if s.project_id == proj]:
+            if s.cram:
+                j = _add_batch_job(hbatch, f'{smdb_proj}/{s.cpg_id}: Move CRAM uploads')
+                s.cram.moved_path = join(new_bucket, f'{s.cpg_id}.cram')
+                s.crai.moved_path = s.cram.moved_path + '.crai'
+                s.cram_md5.moved_path = s.cram.moved_path + '.md5'
+                _move(j, s.cram)
+                _move(j, s.crai)
+                _move(j, s.cram_md5)
+
+        logger.info(f'{proj}: adding jobs to move GVCF uploads')
+        new_bucket = SOURCES['GVCF'].get_staging_bucket(stack, namespace)
+        for s in [s for s in samples if s.project_id == proj]:
+            if s.gvcf:
+                j = _add_batch_job(hbatch, f'{smdb_proj}/{s.cpg_id}: Move GVCF uploads')
+                s.gvcf.moved_path = join(new_bucket, f'{s.cpg_id}.g.vcf.gz')
+                s.tbi.moved_path = s.gvcf.moved_path + '.tbi'
+                _move(j, s.gvcf)
+                _move(j, s.tbi)
+
+    hbatch.run(wait=True, dry_run=dry_run)
```

Comment:
Why `wait=true`, this could cause re-runs.

Reply:
What do you mean? Why would it cause re-runs in contrast with wait=False?

Actually `wait=True` is not really necessary here, because it's the last line in the command